students/srepking/Lesson04/mailroom2.py

```diff
@@ -1,40 +1,42 @@
+import os
+
 def thank_you():
     print('\n''Ok, you chose 1, lets see what we can do.')
     while True:
         name = input('\n''Type the full name to add entry, '
                      '\'list\' to see list of names, or \'e\' to exit: > ')
-        # See if they want a list of names.
+        if name == 'e':
+            return
         if name != 'list':
             break
         for x in donors:
             print(x)
     # Add Name to list if not in list
-    if name not in donors and name != 'e':
+    if name not in donors:
         print('Name is not there, added to the list')
         donors[name] = []
-    if name != 'e':
         amount = input('\n''What is the donation amount? or \'e\' to exit >')
-        if amount != 'e':
-            new_amount = [float(amount)]
-            old_amount = donors[name]
-            donors[name] = old_amount + new_amount
-            print('Thank you so much for the generous gift of ${0:.2f}, {1}!'
-                  .format(float(amount), name))
+        if amount == 'e':
+            return
+        new_amount = [float(amount)]
+        old_amount = donors[name]
+        donors[name] = old_amount + new_amount
+        print('Thank you so much for the generous gift of ${0:.2f}, {1}!'
+              .format(float(amount), name))
```

Comment:
the code is missing feature for adding donations to existing donors. 

Also, suggestion to simplify:
```
amount = float(input("..."))
# add validation here for amount if you'd like
if name in donors:
    donors[name].append(amount)
else:
    donors[name] = [amount]

print("thank you...")

```

Reply:
I implemented the suggestion 'donors[name].append(float(amount))', and then caught the exception instead of using if/else.

Reply:
👍 

there is a great library to make this even easier:
https://www.accelebrate.com/blog/using-defaultdict-python/

I don't mean keep hammering on your code but you seem eager to learn things 🙂

Reply:
Hi Natasha, 
I looked over the default dictionary link you sent, and I see how I could have used it to create a blank entry for a key with no value instead of the if/else loop above. However, I kind of went a different  direction in the last mailroom exercise for Lesson05 and am handling this with an exception now, so that if someone tries to enter a new name, and exception is caught, and the user is prompted to verify that this is what they want to do. Is that OK? or is there still a way I could make this better with the defaultdic library? I am not seeing it! Thanks!